examples/classic_controllers/classic_controllers_dc_motor_test.py

```diff
@@ -1,77 +1,56 @@
+    modelling = 'Cont'
+
+    motor = modelling + '-' + control_type + 'C-' + motor_type + '-v0'
+
+    if motor_type in ['PermExDc', 'SeriesDc']:
+        states = ['omega', 'torque', 'i', 'u']
+    elif motor_type == 'ShuntDc':
+        states = ['omega', 'torque', 'i_a', 'i_e', 'u']
+    elif motor_type == 'ExtExDc':
+        states = ['omega', 'torque', 'i_a', 'i_e', 'u_a', 'u_e']
     else:
-        ref_gen = []
-        for r in ref:
-            ref_gen.append(rg.SwitchedReferenceGenerator(
-                    sub_generators=[
-                    rg.SinusoidalReferenceGenerator(reference_state=r, amplitude_range=(0, 0.3), offset_range=(0, 0.2)),
-                    rg.WienerProcessReferenceGenerator(reference_state=r, amplitude_range=(0, 0.3),
-                                                       offset_range=(0, 0.2)),
-                    rg.StepReferenceGenerator(reference_state=r, amplitude_range=(0, 0.3), offset_range=(0, 0.2))
-                    ], p=[0.5, 0.25, 0.25], super_episode_length=(10000, 100000)
-                )
-            )
-        reference_generator = rg.MultipleReferenceGenerator(ref_gen)
+        raise KeyError(motor_type + ' is not available')
 
-    limit_values = dict(omega=100*np.pi/30)
-    matplotlib.use('TkAgg')
-    env = gem.make(
-        #'DcExtExDisc-v1', visualization=MotorDashboard(state_plots_extended=['omega', 'torque', 'i_a', 'i_e', 'u_a', 'u_e'], visu_period=1),
-        #'DcPermExDisc-v1', visualization=MotorDashboard(state_plots_extended=['omega', 'torque', 'i', 'u'], visu_period=1),
-        #'DcSeriesDisc-v1', visualization=MotorDashboard(state_plots_extended=['omega', 'torque', 'i', 'u'], visu_period=1),
-        #'DcShuntDisc-v1', visualization=MotorDashboard(state_plots_extended=['omega', 'torque', 'i_a', 'i_e', 'u'], visu_period=1),
+    external_ref_plots = [ExternallyReferencedStatePlot(state) for state in states]
 
-        #'DcExtExCont-v1', visualization=MotorDashboard(state_plots_extended=['omega', 'torque', 'i_a', 'i_e', 'u_a', 'u_e'], visu_period=1),
-        'DcPermExCont-v1', visualization=MotorDashboard(state_plots_extended=['omega', 'torque', 'i', 'u'], visu_period=1),
-        #'DcSeriesCont-v1', visualization=MotorDashboard(state_plots_extended=['omega', 'torque', 'i', 'u'], visu_period=1),
-        #'DcShuntCont-v1', visualization=MotorDashboard(state_plots_extended=['omega', 'torque', 'i_a', 'i_e', 'u'], visu_period=1),
-
-        ode_solver='scipy.solve_ivp', solver_kwargs=dict(),
-        #limit_values=limit_values,
-        #load=ConstantSpeedLoad(omega_fixed=50 * np.pi / 30),
-
-        reference_generator=reference_generator
-    )
-    controller = Controller.make(env)
+    matplotlib.use('TkAgg')
+    env = gem.make(motor, visualization=MotorDashboard(additional_plots=external_ref_plots))
 
-    steps = 10001 if type(env.action_space) == Box else 100001
+    controller = Controller.make(env, external_ref_plots=external_ref_plots)
+    steps = 10001
```

Comment:
The default time-Axis width in the MotorDashboard are 10000 steps. If you simulate less than 10k steps, the window will stay partly blank. However, the window should slide, if you simulate for a longer time. 

If desired, the time-Axis width can be adjusted by the user.

Reply:
in my case, it did not slide for more steps -> time length was always fixed to 0.1 seconds in all dc and PMSM use cases I have tested. 

Isn't it possible to formulate a maximum step number which is depicted in a figure, and if the simulated step number is lower that the t axis is reduced automatically?

Reply:
During the last Update of the Dashboard, I already experimented with  "growing" plots.
To me, it looked confusing because the points were moving at each plot update.

Furthermore, a simulation of less then 10k points is only for test cases. And if you keep the figure open with the plt.show(block=True) command, you can zoom to the desired window.

The problem that your plot is not sliding originates probably from your default matplotlib backend. Which one do you typically use? And can you try to execute a script from an (anaconda) shell instead of from PyCharm?